### PR TITLE
Bring your own LIWC & matplotlib dependency fix

### PR DESCRIPTION
**Pull request by sundy1994**
# Basic Info
What's this pull request about? 
Added bring your own LIWC feature #281 and fix matplotlib error during installation #319 .

## BYOL
Now users can upload their own LIWC dictionary like this:
```
testing_chat = FeatureBuilder(
    input_df = chat_df,
    ...
    custom_liwc_dictionary_path = 'my_path/liwc_2015.dic'
)
testing_chat.featurize()
```
If the custom dictionary is provided, `liwc_features()` will call `get_liwc_count()` again on it. The new columns will be named as 'xx_lexical_wordcount_custom' to prevent overwrite existing 'xx_lexical_wordcount' cols from the pickle dic.

It's worth noting that the liwc_2015 dictionary contains emojis like ':)' and ':('. `re.findall()` will throw a parentheses unbalanced error for them, so I add a backslash before all single parentheses in the dic terms while reading the custom dictionary.

## matplotlib issue
"matplotlib>=3.0.0" is added to both`requirements.txt` and `project.toml`. Now our package can be installed to an empty environment as before.

### CODE REVIEW

File: src/team_comm_tools/feature_builder.py [hunk 1]
```diff
@@ -501,6 +522,7 @@ def featurize(self) -> None:
             print("All Done!")
             
             # Store column names of what we generated, so that the user can easily access them
+            # TODO --- this needs to be updated if the user brings their own LIWC, because the custom LIWC features are not in `self.chat_features`.
```
Comment (xehu) [inline src/team_comm_tools/feature_builder.py:525]: @sundy1994 Can you please update the `chat_features` property of the FeatureBuilder object whenever the user generates custom LIWC features? Otherwise, those new column names will not show up when the user tries to check the names of the chat features.

**Comment by xehu** [comment]
# TODO: Please check emoji/word boundary rules and run tests on custom LIWC categories

Emily's sample testing code: [test_byol.ipynb.zip](https://github.com/user-attachments/files/17479437/test_byol.ipynb.zip)

I ran a few simple test cases on the new LIWC, and I found that the emojis (perhaps due to word boundary rules) are not fully working as expected; emojis like `:)`, `:(`, `;)`, etc., which are listed in the dictionary, do not appear to be showing up in the counts.

More generally, would it be possible to read in the dictionary and generate some test cases? For example, if a category consists of "happy, good, great," let's generate a sentence that samples a combination of words with replacement from the category, and words that are not in the category (e.g., "foo"):

> happy happy foo foobar good good great

And we can therefore assert the expected value to be the number of category words we sampled: 5 (in this case).

It should be relatively easy to do this locally with the dictionary, and confirm that we pass all tests for both the old and new dictionaries.

Additionally, let's try variations with punctuation, just to confirm that the new parentheses rules aren't breaking anything. I think it's important to just have this sanity check that everything works as we expect!

**Review by xehu — CHANGES_REQUESTED** [review]
Please make the following changes:

1. Add the custom LIWC feature names to `self.chat_features` in the FeatureBuilder;
2. Add tests to confirm that all the features work as expected, especially with edge cases (e.g., emojis, other punctuation).

Thanks so much for your amazing progress and persistence on this!!

File: src/team_comm_tools/feature_builder.py (hunk 1 above)
Comment (xehu) [reply src/team_comm_tools/feature_builder.py]: DONE

**Review by xehu — APPROVED** [review]
LGTM!!
